core/src/rpc/aggregator.rs

```diff
@@ -707,6 +717,117 @@ mod tests {
+        let verifier = Verifier::new(rpc, config.clone()).await.unwrap();
+        tracing::info!("verifier config: {:#?}", verifier.config);
+        let verifier_wpks = verifier
+            .db
+            .get_watchtower_winternitz_public_keys(None, 0, 0)
+            .await
+            .unwrap();
+        let verifier_challenge_addresses_0 = verifier
+            .db
+            .get_watchtower_challenge_addresses(None, 0, 0)
+            .await
+            .unwrap();
+        let verifier_challenge_addresses_1 = verifier
+            .db
+            .get_watchtower_challenge_addresses(None, 1, 0)
+            .await
+            .unwrap();
+        let verifier_challenge_addresses_2 = verifier
+            .db
+            .get_watchtower_challenge_addresses(None, 2, 0)
+            .await
+            .unwrap();
+        let verifier_challenge_addresses_3 = verifier
+            .db
+            .get_watchtower_challenge_addresses(None, 3, 0)
+            .await
+            .unwrap();
+        tracing::info!(
+            "watchtower_challenge_addresses length: {:?}",
+            watchtower_challenge_addresses.len()
+        );
+        tracing::info!(
+            "verifier_challenge_addresses length: {:?}",
+            verifier_challenge_addresses_0.len()
+        );
+        assert_eq!(
+            config.num_kickoffs_per_sequential_collateral_tx
+                * config.num_kickoffs_per_sequential_collateral_tx,
+            verifier_wpks.len()
+        );
+        assert_eq!(
+            config.num_kickoffs_per_sequential_collateral_tx
+                * config.num_kickoffs_per_sequential_collateral_tx,
+            verifier_challenge_addresses_0.len()
+        );
+        tracing::info!(
+            "watchtower_challenge_addresses: {:?}",
+            watchtower_challenge_addresses
+        );
+        tracing::info!(
+            "verifier_challenge_addresses_0: {:?}",
+            verifier_challenge_addresses_0
+        );
+        tracing::info!(
+            "verifier_challenge_addresses_1: {:?}",
+            verifier_challenge_addresses_1
+        );
+        tracing::info!(
+            "verifier_challenge_addresses_2: {:?}",
+            verifier_challenge_addresses_2
+        );
+        tracing::info!(
+            "verifier_challenge_addresses_3: {:?}",
+            verifier_challenge_addresses_3
+        );
+        assert!(
+            watchtower_wpks[0..config.num_kickoffs_per_sequential_collateral_tx
+                * config.num_kickoffs_per_sequential_collateral_tx]
+                .to_vec()
+                == verifier_wpks,
+            "Winternitz keys of watchtower and verifier are not equal"
+        );
+        assert!(
+            watchtower_challenge_addresses[0..config.num_kickoffs_per_sequential_collateral_tx
+                * config.num_kickoffs_per_sequential_collateral_tx]
+                .to_vec()
+                == verifier_challenge_addresses_2,
```

Comment:
I'm okay with the merge. We should add a caveat comment referencing 
https://github.com/chainwayxyz/clementine/issues/478
```suggestion
                == verifier_challenge_addresses_2, // Caveat: https://github.com/chainwayxyz/clementine/issues/478
```

Reply:
Done